awx/main/dispatch/reaper.py

```diff
@@ -11,6 +11,9 @@
 
 
 def reap_job(j, status):
+    if UnifiedJob.objects.get(id=j.id).status not in ('running', 'waiting'):
+        # just in case, don't reap jobs that aren't running
+        return
```

Comment:
I don't quite see how these would have been returned from the queryset in `reap` (although I might not have stared at that long enough). So is this fixing a race condition?

Reply:
So even `if isinstance(j, ProjectUpdate) and j.launch_type == 'sync'`, we still intend to reap it?

...yes, I think that makes sense. We have basically 2 entries in the database for the same celery ID, and it's super important that we do mark them both as failed, because the parent job context doesn't exist to propagate the failure to the sync (or vice versa).